File: src/CSharp/165.CompareVersionNumbers.cs

using System;
using System.Text;
using System.Collections.Generic;

namespace PeefyLeetCode.CompareVersionNumbers
{
public class Solution {
    public int CompareVersion(string version1, string version2) {
        var v1 = version1.Split(".");
        var v2 = version2.Split(".");
        int n = Math.Max(v1.Length, v2.Length);
        for (int i = 0; i < n; i++) {
            int v1_int = i < v1.Length ? int.Parse(v1[i]) : 0;
            int v2_int = i < v2.Length ? int.Parse(v2[i]) : 0;
            if (v1_int == v2_int) continue;
            if (v1_int > v2_int) return 1;
            else return -1;
        }
        return 0;
    }
}
}

File: src/Python/165.CompareVersionNumbers.py

class Solution:
    def compareVersion(self, version1: str, version2: str) -> int:
        for x, y in itertools.zip_longest(version1.split("."), version2.split("."), fillvalue=0):
            if int(x) != int(y): return 1 if int(x) > int(y) else -1
        return 0


if __name__ == "__main__":
    solution = Solution()
    print(solution.compareVersion("0.1", "1.1"))
